src/systems/audio-debug-system.js

```diff
@@ -0,0 +1,323 @@
+import { THREE } from "aframe";
+
+const MAX_DEBUG_SOURCES = 64;
+
+const CONE_VERTEX = `
+  precision highp float;
+  precision highp int;
+  
+  varying vec2 vUv;
+  varying vec3 vNormal;
+  void main()
+  {
+    vNormal = normalize(normal);
+    gl_Position = projectionMatrix * modelViewMatrix * vec4( position, 1.0 );
+    vUv = position.xz;
+  }
+`;
+const CONE_FRAG = `
```

Comment:
Just FYI our webpack is tooled out so you can make these external like `import mediaHighlightFrag from "./media-highlight-frag.glsl";` Not a huge deal to inline bu this one is reasonably long so it might not hurt. You would need to make MAX_DEBUG_SOURCES a define instead (ShaderMaterial will handle inserting it for you)

Reply:
Great, that's much cleaner. Refactored.